File: solutions/go/raindrops/2/raindrops.go
package raindrops

import (
	"fmt"
	"strings"
)

func Convert(number int) string {
	var b strings.Builder

	if number % 3 == 0 {
		b.WriteString("Pling")
	}
	if number % 5 == 0 {
		b.WriteString("Plang")
	}
	if number % 7 == 0 {
		b.WriteString("Plong")
	}

	output := b.String()

	if len(output) == 0 {
		return fmt.Sprint(number)
	} else {
		return output
	}
}

File: solutions/ruby/raindrops/2/raindrops.rb
class Raindrops
  SOUNDS = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong'}.freeze
  
  def self.convert(number)
    result = SOUNDS.select{ |key, val| (number % key).zero? }.values.join
    result.empty? ? number.to_s : result
  end
end
